core/src/main/java/com/linecorp/armeria/server/logging/ContentPreviewingService.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging;
+
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestLogAccess;
+import com.linecorp.armeria.internal.logging.ContentPreviewerConfigurator;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingHttpService;
+
+/**
+ * Decorates an {@link HttpService} to preview the content of {@link Request}s and {@link Response}s.
+ *
+ * <p>Note that this decorator just sets {@link RequestLog#requestContentPreview()} and
+ * {@link RequestLog#responseContentPreview()}. You can get the previews using {@link RequestLogAccess}.
+ *
+ * <pre>{@code
+ * RequestLogAccess logAccess = ctx.log();
+ * logAccess.whenAvailable(RequestLogProperty.REQUEST_CONTENT_PREVIEW,
+ *                         RequestLogProperty.RESPONSE_CONTENT_PREVIEW).thenApply(...);
```

Comment:
How about using `whenComplete()` to promote a good practice?

Reply:
That's a good suggestion!